clang/test/SemaCUDA/float128.cu

```diff
@@ -0,0 +1,9 @@
+// CPU-side compilation on x86 (no errors expected).
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -aux-triple nvptx64 -x cuda -fsyntax-only -verify %s
+
+// GPU-side compilation on x86 (no errors expected)
+// RUN: %clang_cc1 -triple nvptx64 -aux-triple x86_64-unknown-linux-gnu -fcuda-is-device -x cuda -fsyntax-only -verify %s
```

Comment:
I'd add a test verifying that we do emit diagnostics if fp128 is used in the GPU code.
It would probably need to be done somewhere in the codegen tests as it will not fire in the syntax-only checks.